### PR TITLE
feat: add error dialog component

### PR DESCRIPTION
**Pull request by domw30**
### What does this do?
- Adds an `ErrorDialog` component

### Why are we making this change?
- To be used as a 'pop-up' to notify the user that they are not a member of the conversation. 

This is a very basic iteration of an `ErrorDialog`. The component can be made more generic/customisable in the future if needed.

#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

Screenshot 2024-01-08 at 15 03 19

### CODE REVIEW

File: src/components/error-dialog/container.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
```
Comment (dalefukami) [inline src/components/error-dialog/container.tsx:1]: Why do we need a container if it's not providing any functionality?

File: src/components/error-dialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export class ErrorDialog extends React.Component<Properties> {
+  render() {
+    return (
+      <div {...cn('')}>
+        <IconButton {...cn('close')} size={40} Icon={IconXClose} onClick={this.props.onClose} />
+
+        <div {...cn('heading-container')}>
+          <div {...cn('header')}>Access Denied</div>
+          <div {...cn('divider')} />
```
Comment (dalefukami) [inline src/components/error-dialog/index.tsx:24]: Do we need a "divider"? Why can't the `header` or the `heading-container` just have the border?

**Review by dalefukami — APPROVED** [review]

File: src/components/error-dialog/container.tsx (hunk 1 above)
Comment (domw30) [reply src/components/error-dialog/container.tsx:1]: I was thinking the same, but wondered if at any point in the future we would be pulling in state or actions if this dialog was to be used in other scenarios. Maybe thinking a little too far ahead for that. I can do a follow up to this PR and remove the container

File: src/components/error-dialog/index.tsx (hunk 2 above)
Comment (domw30) [reply src/components/error-dialog/index.tsx:24]: Yeah I can update one of the header borders instead - admittedly this was just a copy and paste from one of the other modals.

Comment (dalefukami) [reply src/components/error-dialog/index.tsx:24]: We definitely need to generalize this modal a bit. We shouldn't need to rebuild this whole thing every time. lol

**Pull request merged**